### PR TITLE
Packets 9 and 139 - Friend Removal

### PR DESCRIPTION
**Pull request by desukuran**
* This packet deals with removing a friend, both ways.

PFireDatabase functions added/modified: QueryUser and RemoveFriend
 * QueryUser has a new overload for an integer arg for the userid. Now you can search users by userid.
 * RemoveFriend, both UserModel args adapted from DeletePendingFriendRequest.

Added Inbound Packet FriendRemoval (9) 
* ! This one has no attribute count from the client.
 * It only expects a userid. From there the server handles the rest.
 * As such, I added an overload to QueryUser to search by userid.

Added Outbound Packet FriendRemoved (139)
 * This one only has a userid, as a confirmation.

### CODE REVIEW

File: src/PFire.Core/Services/PFireDatabase.cs [hunk 1]
```diff
@@ -17,12 +17,14 @@ internal interface IPFireDatabase
         Task InsertMutualFriend(UserModel me, UserModel them);
         Task InsertFriendRequest(UserModel me, UserModel them, string message);
         Task<UserModel> QueryUser(string username);
+        Task<UserModel> QueryUser(int userid);
```
Comment (darcymiranda) [inline src/PFire.Core/Services/PFireDatabase.cs]:
```suggestion
        Task<UserModel> QueryUser(int userId);
```

File: src/PFire.Core/Protocol/Messages/Inbound/FriendRemoval.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using System.Threading.Tasks;
+
+namespace PFire.Core.Protocol.Messages.Inbound
+{
+    internal sealed class FriendRemoval : XFireMessage
+    {
+        public FriendRemoval() : base(XFireMessageType.FriendRemoval) { }
+
+        [XMessageField("userid")]
+        public int userid { get; set; }
```
Comment (darcymiranda) [inline src/PFire.Core/Protocol/Messages/Inbound/FriendRemoval.cs]:
```suggestion
        public int UserId { get; set; }
```

File: src/PFire.Core/Protocol/Messages/Inbound/FriendRemoval.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+        public override async Task Process(IXFireClient context)
+        {
+            var friend = context.Server.Database.QueryUser(userid);
+            await context.Server.Database.RemoveFriend(context.User, friend.Result);
+            await context.SendAndProcessMessage(new FriendRemoved(userid));
```
Comment (darcymiranda) [inline src/PFire.Core/Protocol/Messages/Inbound/FriendRemoval.cs]:
```suggestion
            var friend = context.Server.Database.QueryUser(UserId);
            await context.Server.Database.RemoveFriend(context.User, friend.Result);
            await context.SendAndProcessMessage(new FriendRemoved(UserId));
```

File: src/PFire.Core/Protocol/Messages/Outbound/FriendRemoved.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Threading.Tasks;
+using PFire.Core.Session;
+
+namespace PFire.Core.Protocol.Messages.Outbound
+{
+    internal sealed class FriendRemoved : XFireMessage
+    {
+        public FriendRemoved(int userId) : base(XFireMessageType.FriendRemoved)
+        {
+            userid = userId;
```
Comment (darcymiranda) [inline src/PFire.Core/Protocol/Messages/Outbound/FriendRemoved.cs]:
```suggestion
            UserId = userId;
```

File: src/PFire.Core/Protocol/Messages/Outbound/FriendRemoved.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        }
+
+        [XMessageField("userid")]
+        public int userid { get; set; }
```
Comment (darcymiranda) [inline src/PFire.Core/Protocol/Messages/Outbound/FriendRemoved.cs]:
```suggestion
        public int UserId { get; set; }
```

File: src/PFire.Core/Services/PFireDatabase.cs [hunk 6]
```diff
@@ -166,6 +168,24 @@ public async Task<UserModel> QueryUser(string username)
...
         }
 
+        public async Task<UserModel> QueryUser(int userid)
+        {
+            using var scope = _serviceProvider.CreateScope();
+            var databaseContext = scope.ServiceProvider.GetRequiredService<IDatabaseContext>();
+
+            return await databaseContext.Set<User>()
+                                        .AsNoTracking()
+                                        .Where(a => a.Id == userid)
```
Comment (darcymiranda) [inline src/PFire.Core/Services/PFireDatabase.cs]:
```suggestion
                                        .Where(a => a.Id == userId)
```

File: src/PFire.Core/Services/PFireDatabase.cs [hunk 7]
```diff
@@ -166,6 +168,24 @@ public async Task<UserModel> QueryUser(string username)
                                         .FirstOrDefaultAsync();
         }
 
+        public async Task<UserModel> QueryUser(int userid)
```
Comment (darcymiranda) [inline src/PFire.Core/Services/PFireDatabase.cs]:
```suggestion
        public async Task<UserModel> QueryUser(int userId)
```

**Review by darcymiranda — COMMENTED** [review]
Looks great! Thank you. My only nitpick is let's keep to C# naming standards (PascelCase for properties, camelCase for parameters)

**Comment by desukuran** [comment]
There, it shouldn't have 7 redundant commits. I merged it into one.

**Pull request merged**